src/main/java/org/springframework/data/neo4j/core/mapping/DefaultRelationshipDescription.java

```diff
@@ -55,13 +56,30 @@ final class DefaultRelationshipDescription extends Association<Neo4jPersistentPr
 		this.relationshipObverse = relationshipObverse;
 		this.type = type;
 		this.dynamic = dynamic;
-		this.source = source;
+		this.source = getSourceForField(source, type, dynamic, target, direction, relationshipProperties);
 		this.fieldName = fieldName;
 		this.target = target;
 		this.direction = direction;
 		this.relationshipPropertiesClass = relationshipProperties;
 	}
 
+	private static NodeDescription<?> getSourceForField(NodeDescription<?> source, String type, boolean dynamic,
+			NodeDescription<?> target, Relationship.Direction direction,
+			@Nullable NodeDescription<?> relationshipProperties) {
+		NodeDescription<?> parent = source.getParentNodeDescription();
+		if (parent == null) {
+			return source;
+		}
+		for (RelationshipDescription relationship : parent.getRelationships()) {
```

Comment:
This solution does not seem to work consistently. I could detect invocations where `parent.getRelationships()` returns an empty list.

@michael-simons / @meistermeier do you have any idea how to get a consistent behavior?

Reply:
making `source` lazy (`Lazy<NodeDescription<?>>`) seems to solve this issue